pkg/config/validation/validation.go

```diff
@@ -2007,8 +2008,13 @@ var ValidateVirtualService = registerValidateFunc("ValidateVirtualService",
 			}
 		}
 
+		isDelegate := false
 		if len(virtualService.Hosts) == 0 {
-			errs = appendErrors(errs, fmt.Errorf("virtual service must have at least one host"))
+			if features.EnableVirtualServiceDelegate {
+				isDelegate = true
+			} else {
+				errs = appendErrors(errs, fmt.Errorf("virtual service must have at least one host"))
+			}
```

Comment:
You also need to ensure that the delegate vs has no gateways section in the top. And has no tcp or tls blocks.

Reply:
make sense